lib/iiif_print/persistence_layer/valkyrie_adapter.rb

```diff
@@ -68,10 +68,24 @@ def self.solr_name(field_name)
         Hyrax.config.index_field_mapper.solr_name(field_name.to_s)
       end
 
-      ##
-      # @todo implement this logic
-      def self.destroy_children_split_from(file_set:, work:, model:)
-        super
+      def self.destroy_children_split_from(file_set:, work:, model:, user:)
+        # look first for children by the file set id they were split from
+        children = Hyrax.query_service.find_all_of_model(model: model).select { |m| m.split_from_pdf_id == file_set.id }
+        if children.blank?
+          # find works where file name and work `to_param` are both in the title
+          children = Hyrax.query_service.find_all_of_model(model: model).select { |m| m.title.include?(file_set.label) && m.title.include?(work.to_param) }
```

Comment:
How might we end up in this situation?  This is another poorly performing query that I'd just as soon not have to call.  And it can easily have false positives.

Reply:
agreed, i was trying to match this version
https://github.com/scientist-softserv/iiif_print/blob/fix-delete-work/lib/iiif_print/persistence_layer/active_fedora_adapter.rb#L86-L102

but removing it sounds great